packages/dev/loaders/src/glTF/2.0/glTFLoader.ts

```diff
@@ -2172,13 +2200,22 @@ export class GLTFLoader implements IGLTFLoader {
                 promises.push(
                     this.loadTextureInfoAsync(`${context}/metallicRoughnessTexture`, properties.metallicRoughnessTexture, (texture) => {
                         texture.name = `${babylonMaterial.name} (Metallic Roughness)`;
-                        babylonMaterial.metallicTexture = texture;
+                        if (this.parent.useOpenPBR) {
+                            (babylonMaterial as OpenPBRMaterial).baseMetalnessTexture = texture;
+                            (babylonMaterial as OpenPBRMaterial).specularRoughnessTexture = texture;
+                            (babylonMaterial as OpenPBRMaterial)._useRoughnessFromMetallicTextureGreen = true;
+                            (babylonMaterial as OpenPBRMaterial)._useMetallicFromMetallicTextureBlue = true;

```

Comment:
The glTF loader tries to make sure all shader variants are accounted for in the first synchronous pass. I'm guessing these flags will cause the shader to be different. If so, these should be moved to below, like with the PBRMaterial flags.

Reply:
Fixed.